txp/Pos/Txp/MemState/Class.hs

```diff
@@ -91,13 +98,15 @@ modifyTxpLocalData f =
         STM.writeTVar txpTip newTip
         STM.writeTVar txpExtra newExtra
         pure res
+ where
+   withLock = L.bracket_ (takeMVar txpLocalDataLock) (putMVar txpLocalDataLock ())
```

Comment:
As I mentioned in #953, this function is instantiated with transformer stack that contains `ExceptT`, hence `MonadMask` won't work here. Perhaps `Mockable Bracket` will.